packages/react-next/src/components/SearchBox/SearchBox.base.tsx

```diff
@@ -1,264 +1,195 @@
-    if (newProps.value !== undefined) {
-      this._latestValue = newProps.value;
-      // If the user passes in null, substitute an empty string
-      // (passing null is not allowed per typings, but users might do it anyway)
-      this.setState({
-        value: newProps.value || '',
-      });
-    }
-  }
-
-  public render() {
-    const {
-      ariaLabel,
-      placeholder,
-      className,
-      disabled,
-      underlined,
-      styles,
-      // tslint:disable-next-line:deprecation
-      labelText,
-      theme,
-      clearButtonProps,
-      disableAnimation,
-      iconProps,
-      id = this._fallbackId,
-    } = this.props;
-    const { value, hasFocus } = this.state;
-    const placeholderValue = placeholder !== undefined ? placeholder : labelText;
-
-    const classNames = getClassNames(styles!, {
-      theme: theme!,
-      className,
-      underlined,
-      hasFocus,
-      disabled,
-      hasInput: value!.length > 0,
-      disableAnimation,
-    });
-
-    const nativeProps = getNativeProps<React.InputHTMLAttributes<HTMLInputElement>>(this.props, inputProperties, [
-      'className',
-      'placeholder',
-      'onFocus',
-      'onBlur',
-      'value',
-    ]);
-
-    return (
-      <div role="search" ref={this._rootElement} className={classNames.root} onFocusCapture={this._onFocusCapture}>
-        <div className={classNames.iconContainer} onClick={this._onClickFocus} aria-hidden={true}>
-          <Icon iconName="Search" {...iconProps} className={classNames.icon} />
-        </div>
-        <input
-          {...nativeProps}
-          id={id}
-          className={classNames.field}
-          placeholder={placeholderValue}
-          onChange={this._onInputChange}
-          onInput={this._onInputChange}
-          onBlur={this._onBlur}
-          onKeyDown={this._onKeyDown}
-          value={value}
-          disabled={disabled}
-          role="searchbox"
-          aria-label={ariaLabel}
-          ref={this._inputElement}
-        />
-        {value!.length > 0 && (
-          <div className={classNames.clearButton}>
-            <IconButton
-              onBlur={this._onBlur}
-              styles={{ root: { height: 'auto' }, icon: { fontSize: '12px' } }}
-              iconProps={{ iconName: 'Clear' }}
-              {...clearButtonProps}
-              onClick={this._onClearClick}
-            />
-          </div>
-        )}
-      </div>
-    );
-  }
-
-  /**
-   * Sets focus to the search box input field
-   */
-  public focus() {
-    if (this._inputElement.current) {
-      this._inputElement.current.focus();
-    }
-  }
+const getClassNames = classNamesFunction<ISearchBoxStyleProps, ISearchBoxStyles>();
+const COMPONENT_NAME = 'SearchBox';
 
-  /**
-   * Returns whether or not the SearchBox has focus
-   */
-  public hasFocus(): boolean {
-    return !!this.state.hasFocus;
+const useComponentRef = (
+  props: ISearchBoxProps,
```

Comment:
Probably should just take in componentRef and not the whole props object, so that it's clear what's used.